From 9a18968253e28ba8d8bdf646731087000c7876b7 Mon Sep 17 00:00:00 2001
From: sclarkson <sc@lambdal.com>
Date: Tue, 21 Mar 2023 21:44:49 +0000
Subject: [PATCH] Fix kDefaultTimeout multiple definition build failure
 (#97270)

Make the namespace explicit to avoid the constexpr conflict on GCC 11.

Fixes #90448

@ezyang

Pull Request resolved: https://github.com/pytorch/pytorch/pull/97270
Approved by: https://github.com/ezyang
---
 torch/csrc/distributed/c10d/ProcessGroupGloo.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/torch/csrc/distributed/c10d/ProcessGroupGloo.hpp b/torch/csrc/distributed/c10d/ProcessGroupGloo.hpp
index a64bc37c4de522..e4d2967c8604ea 100644
--- a/torch/csrc/distributed/c10d/ProcessGroupGloo.hpp
+++ b/torch/csrc/distributed/c10d/ProcessGroupGloo.hpp
@@ -125,7 +125,7 @@ class TORCH_API ProcessGroupGloo : public Backend {
     }
 
     void wait(const std::vector<std::string>& keys) override {
-      store_->wait(keys, Store::kDefaultTimeout);
+      store_->wait(keys, ::c10d::Store::kDefaultTimeout);
     }
 
     void wait(
